server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java

```diff
@@ -138,12 +138,9 @@ public void init() throws ServletException
   {
     super.init();
 
-    // separate client with more aggressive connection timeouts
-    // to prevent cancellations requests from blocking queries
+    // Note that httpClinetProvider is setup to return same HttpClient instance on each get() so
+    // it is same http client as that is used by parent ProxyServlet.
     broadcastClient = httpClientProvider.get();
-    broadcastClient.setConnectTimeout(CANCELLATION_TIMEOUT_MILLIS);
```

Comment:
CANCELLATION_TIMEOUT_MILLIS timeout is also used at request time https://github.com/druid-io/druid/blob/master/server/src/main/java/io/druid/server/AsyncQueryForwardingServlet.java#L190

that said, connectTimeout relates to the socket timeout and not sure if request timeout would cover that case. Given that there is only one HttpClient instance, we can't have two different connect timeouts . I am considering making connectTimeout configurable and default value be 500 ms as it was before. (jetty default is 15000 ms )
Or, instead of current change, we could change the provider to be non singleton and broadcastClient really be a different instance, but I don't want to do that as understanding resources is simpler with one HttpClient instance with one queue and threadpool rather than two.